src/bsoncxx/array/view.cpp

```diff
@@ -30,7 +30,7 @@ namespace array {
 
 namespace {
 bson_iter_t to_bson_iter_t(element e) {
-    bson_iter_t i;
+    bson_iter_t i{};
```

Comment:
This should silence coverity, but more work is required, see https://jira.mongodb.org/browse/CXX-1499

Reply:
weird, they should be equivalent?

Reply:
In C++ yes, but bson_iter_t is a C struct, so a POD. Fields are not default initialized, so we get whatever garbage is on the stack here in any fields that we don't immediately set. This at least zero-inits any we don't explicitly set. That is still probably wrong, but less so, hence CXX-1499